### PR TITLE
RFC33: ckb vm version1 changes

### PR DESCRIPTION
**Pull request by mohanson**

### CODE REVIEW

**Comment by doitian** [comment]
See #235, it's planned to use PR number as the RFC number. If you have multiple RFCs to submit, please split it into multiple PRs.

File: rfcs/0029-ckb-hardfork/0029-ckb-hardfork.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+Author: Wanbiao Ye
+Organization: Nervos Foundation
+Created: 2021-05-25
+---
+
+# CKB Hardfork
+
+## Abstract
+
+This document describes the first hard fork of ckb.
```
Comment (doitian) [inline rfcs/0029-ckb-hardfork/0029-ckb-hardfork.md:14]: This document only describes the syscalls upgrade, not the first fork of CKB.

Comment (doitian) [reply rfcs/0029-ckb-hardfork/0029-ckb-hardfork.md:14]: It's better to mention the related RFCs and add the backlink to the related RFCs so when developers read the old RFC, they known there are updates to the syscalls.

File: rfcs/0028-ckb-vm-version-1/0028-ckb-vm-version-1.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
+---
+Number: "0028"
+Category: Standards Track
+Status: Proposal
+Author: Wanbiao Ye
+Organization: Nervos Foundation
+Created: 2021-05-25
+---
+
+# VM version1
```
Comment (doitian) [inline rfcs/0028-ckb-vm-version-1/0028-ckb-vm-version-1.md:10]: It's better to mention the RFC0003 and add the backlink in RFC0003 to this RFC so when developers read RFC0003, they known there's a new version of VM.

File: rfcs/0029-ckb-hardfork/0029-ckb-hardfork.md (hunk 1 above)
Comment (mohanson) [reply rfcs/0029-ckb-hardfork/0029-ckb-hardfork.md:14]: > This document only describes the syscalls upgrade, not the first fork of CKB.

This is the @xxuejie 's suggestion. He hopes that there will be an RFC describing all the modifications of hardfork, syscalls is only a part of it.

Comment (xxuejie) [reply rfcs/0029-ckb-hardfork/0029-ckb-hardfork.md:14]: But for now, you can still describe it as VM & syscall changes for the next hardfork. Later we can expand this to cover all CKB changes(and change the wording here), or use a separate RFC

File: rfcs/0000-ckb-vm-version-1/0000-ckb-vm-version-1.md [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ---
-Number: "0028"
-Category: Standards Track
-Status: Proposal
+Number: "0000"
```
Comment (doitian) [inline rfcs/0000-ckb-vm-version-1/0000-ckb-vm-version-1.md:2]: Use "0236"

**Comment by doitian** [comment]
Please mention #232 how the node selects VM versions.

**Comment by doitian** [comment]
For vm version 1 in CKB, one special rule can be added. If the loaded ELF has an [ELF section](https://refspecs.linuxfoundation.org/elf/gabi4+/ch4.eheader.html) named “.ckb.forks”, CKB-VM will skip executing the binary, and just treat the program as always success script. The consumed cycles are just the ones used to load the script and boot the VM. CKB nodes of the next fork version (ckb2021) should not relay transactions with ".ckb.forks" section because they cannot verify it.

**Comment by doitian** [comment]
Please change RFC numbers according to https://github.com/nervosnetwork/rfcs/issues/246 . Sorry for the inconvenience.

**Review by quake — APPROVED** [review]

**Review by doitian — APPROVED** [review]

**Review by doitian — APPROVED** [review]

**Comment by doitian** [comment]
Assigned reviewers: @quake @doitian

**Review by quake — APPROVED** [review]

**Pull request merged**